File: src/ui-client/src/models/Help/HelpPages.ts
/* Copyright (c) 2020, UW Medicine Research IT, University of Washington
 * Developed by Nic Dobbins and Cliff Spital, CRIO Sean Mooney
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */ 

export interface HelpPages {
    id: number;
    categoryId: number;
    title: string;
}

// export interface HelpPagesDTO extends HelpPages {

// }

export interface HelpPageContent {
    id: number;
    pageId: number;
    orderId: number;
    type: string;
    textContent: string;
    imageContent: Uint16Array;
    imageId: string;
}

// export interface HelpPageContentDTO extends HelpPageContent {

// }

// export const HelpPagesDTO = (helpPages: HelpPages): HelpPagesDTO => {
//     return {
//         id: helpPages.id,
//         categoryId: helpPages.categoryId,
//         title: helpPages.title
//     }
// }

// export const HelpPageContentDTO = (helpPageContent: HelpPageContent): HelpPageContentDTO => {
//     return {
//         id: helpPageContent.id,
//         pageId: helpPageContent.pageId,
//         orderId: helpPageContent.orderId,
//         type: helpPageContent.type,
//         textContent: helpPageContent.textContent,
//         imageContent: helpPageContent.imageContent,
//         imageId: helpPageContent.imageId
//     }
// }

File: src/server/Model/Help/HelpPages.cs
// Copyright (c) 2020, UW Medicine Research IT, University of Washington
// Developed by Nic Dobbins and Cliff Spital, CRIO Sean Mooney
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.
using System.Data.Common;
using System.Collections.Generic;
using System.Threading.Tasks;
using Model.Compiler;
using Microsoft.Extensions.Logging;
using Model.Error;
namespace Model.Help
{
    public class HelpPages
    {
        public interface IHelpPages
        {
            Task<IEnumerable<HelpPageSql>> GetAllPagesAsync();
            Task<IEnumerable<HelpPageContentSql>> GetPageContentAsync(int pageid);
            Task<IEnumerable<HelpPageCategorySql>> GetHelpPageCategoriesAsync();
        }

        readonly IHelpPages help;
        readonly ILogger<HelpPages> log;

        public HelpPages(IHelpPages help, ILogger<HelpPages> logger)
        {
            this.help = help;
            log = logger;
        }

        public async Task<IEnumerable<HelpPageSql>> GetAllPagesAsync()
        {
            log.LogInformation("Getting all help pages.");
            try
            {
                return await help.GetAllPagesAsync();
            }
            catch (DbException de)
            {
                log.LogError("Failed to get all help pages. Error:{Error}", de.Message);
                de.MapThrow();
                throw;
            }
        }

        public async Task<IEnumerable<HelpPageCategorySql>> GetHelpPageCategoriesAsync()
        {
            log.LogInformation("Getting all help categories.");
            try
            {
                return await help.GetHelpPageCategoriesAsync();
            }
            catch (DbException de)
            {
                log.LogError("Failed to get help page categories. Error:{Error}", de.Message);
                de.MapThrow();
                throw;
            }
        }

        public async Task<IEnumerable<HelpPageContentSql>> GetPageContentAsync(int pageid)
        {
            log.LogInformation("Getting all help pages.");
            try
            {
                return await help.GetPageContentAsync(pageid);
            }
            catch (DbException de)
            {
                log.LogError("Failed to get all help pages. PageId:{PageId} Error:{Error}", pageid, de.Message);
                de.MapThrow();
                throw;
            }
        }
    }
}
